ci-operator/step-registry/openshift/e2e/test/clusterinfra-qe/openshift-e2e-test-clusterinfra-qe-commands.sh

```diff
@@ -0,0 +1,344 @@
+if [[ $IS_ACTIVE_CLUSTER_OPENSHIFT != "false" ]]; then
+    oc wait clusteroperators --all --for=condition=Progressing=false --timeout=15m
+    oc get clusterversion version -o yaml || true
+fi
+
+# execute the cases
+function run {
+    cd /go/src/github.com/openshift/cluster-api-actuator-pkg
+    # failures happening after this point should not be caught by the Overall CI test suite in RP
+    touch "${ARTIFACT_DIR}/skip_overall_if_fail"
+    ret_value=0
+    set -x
+    echo "TEST_FILTERS_CLUSTERINFRASTRUCTURE: \"${TEST_FILTERS_CLUSTERINFRASTRUCTURE:-}\""
+    if [[ -n "$TEST_FILTERS_CLUSTERINFRASTRUCTURE" ]]; then
+        hack/ci-integration.sh --junit-report=junit_cluster-api-actuator-testutils.xml --output-dir=/logs/artifacts/ --label-filter="${TEST_FILTERS_CLUSTERINFRASTRUCTURE}" -p || ret_value=$?
+    else
+        hack/ci-integration.sh --junit-report=junit_cluster-api-actuator-testutils.xml --output-dir=/logs/artifacts/ --label-filter='!disruptive' -p || ret_value=$?
+    fi
+    set +x
+    set +e
+    echo "try to handle result"
+    handle_result
+    echo "done to handle result"
+    if [ "W${ret_value}W" == "W0W" ]; then
+        echo "success"
+    else
+        echo "fail"
+    fi
+
+    # summarize test results
+    echo "Summarizing test results..."
+    if ! [[ -d "${ARTIFACT_DIR:-'/default-non-exist-dir'}" ]]; then
+        echo "Artifact dir '${ARTIFACT_DIR}' not exist"
+        exit 0
+    else
+        echo "Artifact dir '${ARTIFACT_DIR}' exist"
+        ls -lR "${ARTIFACT_DIR}"
+        files="$(find "${ARTIFACT_DIR}" -name '*.xml' | wc -l)"
+        if [[ "$files" -eq 0 ]]; then
+            echo "There are no JUnit files"
+            exit 0
+        fi
+    fi
+    declare -A results=([failures]='0' [errors]='0' [skipped]='0' [tests]='0')
+    input_file="/tmp/zzz-tmp.log"
+    grep -r -E -h -o 'testsuite.*tests="[0-9]+"[^>]*' "${ARTIFACT_DIR}" >"$input_file" || exit 0
+    while read row; do
+        for ctype in "${!results[@]}"; do
+            count="$(sed -E "s/.*$ctype=\"([0-9]+)\".*/\1/" <<<$row)"
+            if [[ -n $count ]]; then
+                let results[$ctype]+=count || true
+            fi
+        done
+    done <"$input_file"
+
+    TEST_RESULT_FILE="${ARTIFACT_DIR}/test-results.yaml"
+    cat >"${TEST_RESULT_FILE}" <<-EOF
+openshift-e2e-test-clusterinfra-qe:
+  total: ${results[tests]}
+  failures: ${results[failures]}
+  errors: ${results[errors]}
+  skipped: ${results[skipped]}
+EOF
+    if [ ${results[failures]} != 0 ]; then
+        echo '  failingScenarios:' >>"${TEST_RESULT_FILE}"
+        # Every failure scenario will have a directory in machine-api-e2e-suite
+        readarray -t failingscenarios < <(find "${ARTIFACT_DIR}/machine-api-e2e-suite/" -mindepth 1 -maxdepth 1 -type d -exec basename {} \; | sort --unique)
+        for ((i = 0; i < ${#failingscenarios[@]}; i++)); do
+            echo "    - ${failingscenarios[$i]}" >>"${TEST_RESULT_FILE}"
```

Comment:
One of the rehearse job failed to get the failingScenarios.

Reply:
Not a blocker of this PR though. If you'd like to get this one in and use a follow up PR to fix that, I'm fine with that, too. Let me know which one do you prefer.

Reply:
@liangxia Done, please take a look when you got time.